src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -420,36 +419,17 @@ def _check_dataloader_iterable(
         )
 
 
-def _worker_check(dataloader: object, using_spawn: bool, name: str) -> None:
+def _worker_check(trainer: "pl.Trainer", dataloader: object, name: str) -> None:
     if not isinstance(dataloader, DataLoader):
         return
 
-    num_cpus = multiprocessing.cpu_count()
-
-    # ddp_spawn + num_workers > 0 don't mix! tell the user
-    if dataloader.num_workers > 0 and using_spawn:
-        if not dataloader.persistent_workers:
-            rank_zero_warn(
-                "num_workers>0, persistent_workers=False, and strategy=ddp_spawn"
-                " may result in data loading bottlenecks."
-                " Consider setting persistent_workers=True"
-                " (this is a limitation of Python .spawn() and PyTorch)"
-            )
-
-    elif dataloader.num_workers == 0 and using_spawn:
-        if not dataloader.persistent_workers:
-            rank_zero_warn(
-                "strategy=ddp_spawn and num_workers=0 may result in data loading bottlenecks."
-                " Consider setting num_workers>0 and persistent_workers=True"
-            )
-
-    elif dataloader.num_workers <= 2 < num_cpus and not using_spawn:
+    upper_bound = suggested_max_num_workers(trainer.num_devices)
+    if dataloader.num_workers <= 2 < upper_bound or dataloader.num_workers < 2 <= upper_bound:
+        # TODO
         # if changed, update the `filterwarnings` snippet in 'speed.html#num-workers'
         rank_zero_warn(
```

Comment:
Since our bounds now match PyTorch's, should we filter https://github.com/pytorch/pytorch/blob/v2.0.1/torch/utils/data/dataloader.py#L488-L563 so that users don't get the warning twice? Alternatively, we could remove ours

Reply:
Do you think you could assert that the dataloader warning is not raised?

Reply:
@awaelchli I just this function a try but `torch` complains about it: it wants half the number of CPUs as num workers (36 not 72). Particularly this condition: https://github.com/pytorch/pytorch/blob/main/torch/utils/data/dataloader.py#L557